engine/runtime-compiler/src/main/scala/org/enso/compiler/Compiler.scala

```diff
@@ -573,7 +573,6 @@ class Compiler(
           context.updateModule(
             module,
             u => {
-              u.resetScope()
```

Comment:
_"resetScope is also obsolete"_ - wow, now you'll be responsible for the deleted code ;-)

The primary use-case for these changes is likely in the IDE and repeated recompilation. If that is working for you without these resets, then let's try it!